packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -304,10 +313,16 @@ private MethodBodyStatement[] BuildPrimaryConstructorBody(IReadOnlyList<Paramete
             {
                 if (f != _apiKeyAuthField
                     && f != EndpointField
-                    && !f.Modifiers.HasFlag(FieldModifiers.Const)
-                    && clientOptionsPropertyDict.TryGetValue(f.Name.ToCleanName(), out var optionsProperty))
+                    && !f.Modifiers.HasFlag(FieldModifiers.Const))
                 {
-                    body.Add(f.Assign(ClientOptionsParameter.Property(optionsProperty.Name)).Terminate());
+                    if (f == _apiVersionField && ClientOptions.VersionProperty != null)
+                    {
+                        body.Add(f.Assign(ClientOptionsParameter.Property(ClientOptions.VersionProperty.Name)).Terminate());
+                    }
+                    else if (clientOptionsPropertyDict.TryGetValue(f.Name.ToCleanName(), out var optionsProperty))
```

Comment:
I wonder why here we need to use `f.Name.ToCleanName`
we should avoid building the name again - here maybe it could be `f.Property.Name`

Reply:
`f` is FieldProvider, there is no Property of FieldProvider. and Field name is not the same as the name of  `ClientOptions` property. In client Field, the field name is `_XXX`, in the clientOptions, the property name is `XXX`.

Reply:
anyhow I think we should avoid changing the name back and forth.
This piece of code is not introduced by you, therefore it is fine if we leave it here.